content/docs/2.17/scalers/temporal.md

```diff
@@ -0,0 +1,108 @@
++++
+title = "Temporal"
+availability = "v2.17+"
+maintainer = "Community"
+description = "Scale applications based on Temporal task queue."
+go_file = "temporal_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `temporal` trigger that scales based on a temporal task queue.
+
+```yaml
+triggers:
+  - type: temporal
+    metadata:
+      namespace: default
+      taskQueue: "workflow_with_single_noop_activity:test"
+      targetQueueSize: "2"
+      activationTargetQueueSize: "0"
+      endpoint: temporal-frontend.temporal.svc.cluster.local:7233
+```
+
+**Parameter list:**
+
+- `endpoint` - This parameter specifies the URL of the Temporal gRPC service. You need to provide the service address in the format `<hostname>:<port>`. (Required)
+- `endpointFromEnv` - Defines the endpoint, similar to the `endpoint` parameter, but the value is read from an environment variable. (Optional)
+- `namespace` - The namespace of the temporal service. (Default:`default`, Optional)
+- `activationTargetQueueSize` - This sets the target value for activating the scaler. More information about activation thresholds can be found  [here](./../concepts/scaling-deployments.md#activating-and-scaling-thresholds). (Default: `0`, Optional)
+- `targetQueueSize` - Target value for queue length passed to the scaler. The scaler will cause the replicas to increase if the queue message count is greater than the target value per active replica. (Default: `5`, Optional)
+- `taskQueue` - This parameter specifies the task queue name. (Required)
+- `queueTypes` - Task Queue type, which can be either workflow or activity. The default type is workflow (Optional)
+- `buildId` - Build IDs identify Worker versions for Workflow versioning and task compatibility (Optional)
+- `selectAllActive` - Include all active versions (Default:`false`, Optional)
+- `selectUnversioned` - Include the unversioned queue (Default:`false`, Optional)
+- `apiKey` - API key for authentication with temporal cloud (Optional)
+- `apiKeyFromEnv` - API key for authentication similer to `apiKey`, but read from an environment variable (Optional)
```

Comment:
Not sure if we should allow `apiKey` here, we should encourage users to use TriggerAuthentication for this kind of data, WDYT?

Reply:
If that's the best practice we follow, I will move to `TriggerAuthentication`. However, I believe we should keep it here to simplify the configuration.

Reply:
That's the usual way we deal with this kind of information in other scalers.

CC @kedacore/keda-maintainers

Reply:
Removed it from the `triggerMetadata`